server/src/main/java/org/opensearch/index/IndexService.java

```diff
@@ -506,6 +506,17 @@ public synchronized IndexShard createShard(
                         }
                         logger.info("DocRep shard {} is migrating to remote", shardId);
                         seedRemote = true;
+                    } else if (sourceNode == null) {
+                        if (routing.primary() == false) {
+                            throw new IllegalStateException(
+                                "Expected the shard to be primary because the source node is null " + routing.shardId()
+                            );
+                        }
+
+                        // seeding when index metadata says so
+                        // ToDo Check for this shard is seeded or not from index metadata
```

Comment:
Lets remove this as we will not do it now, since seeding is anyways incremental for process reboots.